services/streammanager/kafka/kafkamanager.go

```diff
@@ -329,7 +332,15 @@ func NewProducer(destination *backendconfig.DestinationT, o common.Opts) (*Produ
 	if err != nil {
 		return nil, err
 	}
-	return &ProducerManager{p: p, timeout: o.Timeout, codecs: codecs}, nil
+
+	// @TODO embedAvroSchemaID should come from control plane (i.e. destination config)
```

Comment:
Shouldn't we add config-based support for this already & let it be overridden through this environment variable for the time being? That way, as soon as control plane is ready with their changes, the only thing we'll need to do is to remove the override.

Reply:
The configuration needs to be reworked anyway and if I understand correctly the whole thing will be picked up by @saurav-malani. He'll make it proper don't worry :+1: